ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryCleanupService.java

```diff
@@ -115,11 +115,14 @@ public String getAppId() {
     public void run() {
       try {
         final Path pathName = Paths.get(stateDir + "/" + appId);
-        FileUtils.deleteDirectory(new File(String.valueOf(pathName.normalize())));
-        LOG.warn("Deleted local state store for non-existing query {}. "
-            + "This is not expected and was likely due to a "
-            + "race condition when the query was dropped before.",
-            appId);
+        final File directory = new File(String.valueOf(pathName.normalize()));
+        if (directory.exists()) {
```

Comment:
While staring at the test output, I noticed that the director sometimes doesn't exist when we enter this block. This check will add a small amount of extra race-condition protection, but we could still have two threads traversing the directory tree at the same time.